File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        last_max = -1
        
        for i in range(len(arr) - 1, -1, -1):
            arr[i], last_max = last_max, max(last_max, arr[i])
        
        return arr
        

File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.js
const replaceElements = (arr) => {
  let lastMax = -1;

  for (let i = arr.length - 1; i >= 0; i--) {
    [arr[i], lastMax] = [lastMax, Math.max(lastMax, arr[i])];
  }

  return arr;
};